front/ui/ui-charts/src/spaceTimeChart/components/TimeCaptions.tsx

```diff
@@ -5,27 +5,33 @@ import { HOUR, MINUTE } from '../lib/consts';
 import { type DrawingFunction } from '../lib/types';
 import { computeVisibleTimeMarkers, getCrispLineCoordinate } from '../utils/canvas';
 import { WHITE_ALPHA_75 } from '../utils/colors';
+import {
+  DATE_OPTIONS,
+  DEFAULT_LOCALE,
+  HOUR_OPTIONS_LONG,
+  HOUR_OPTIONS_SHORT,
+  MINUTE_OPTIONS,
+} from '../utils/dates';
 
 const MARGIN = 100;
-const MINUTES_FORMATTER = (t: number) => `:${new Date(t).getMinutes().toString().padStart(2, '0')}`;
+
+const MINUTES_FORMATTER = (t: number) => {
+  const formatted = new Intl.DateTimeFormat(DEFAULT_LOCALE, MINUTE_OPTIONS).format(t);
+  const minutes = formatted.slice(-2);
+  return `:${minutes}`;
+};
+
 const HOURS_FORMATTER = (t: number, pixelsPerMinute: number) => {
-  const date = new Date(t);
   if (pixelsPerMinute > 1) {
-    const hours = date.getHours().toString().padStart(2, '0');
-    const minutes = date.getMinutes().toString().padStart(2, '0');
-    return `${hours}:${minutes}`;
+    return new Intl.DateTimeFormat(DEFAULT_LOCALE, HOUR_OPTIONS_LONG).format(t);
   } else {
-    return date.getHours().toString().padStart(2, '0');
+    const formatted = new Intl.DateTimeFormat(DEFAULT_LOCALE, HOUR_OPTIONS_SHORT).format(t);
+    return formatted.slice(0, 2);
```

Comment:
Why do we need to `slice()` here? Doesn't the string only contain the hours and nothing else already?